### PR TITLE
fix the bug that in the new version of openshift used clusterresource…

### PR DESCRIPTION
**Pull request by BlueShells**
fix the bug that in the new version of openshift used clusterresourcequota

### CODE REVIEW

File: scripts/monitoring/cron-send-usage-pv.py [hunk 1]
```diff
@@ -116,13 +152,23 @@ def main():
...
+        usage_pv = (convert_to_bytes(pv_used)*100)/convert_to_bytes(cluster_capacity_max)
+        logger.debug("percent of usage of pv: %s", usage_pv)
+        logger.debug("datasend to zabbix:")
+        logger.debug("datasend to zabbix: max_gb %s",convert_to_gb(cluster_capacity_max))
+        logger.debug("datasend to zabbix:pv used: %s",convert_to_gb(pv_used))
+        send_metrics(usage_pv, convert_to_gb(cluster_capacity_max), convert_to_gb(pv_used))
     else:
         cluster_capacity_max = get_max_capacity(dynamic_pod_name)
         logger.debug("cluster_capacity_max: %s", cluster_capacity_max)
         pv_used = get_pv_usage()
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-usage-pv.py:167]: @BlueShells, does this pv_used need to be converted?

File: scripts/monitoring/cron-send-usage-pv.py [hunk 2]
```diff
@@ -102,6 +102,42 @@ def get_pv_usage():
             capacity_int = int(capacity.strip('GIgi'))
             total = total + capacity_int
     return total
+def get_pv_usage_clusterresourcequota():
+    """get all the pv used if the cluster is using the clusterresourcequota"""
+    pv_info = runOCcmd_yaml(" get clusterresourcequota/persistent-volume ")
+
+
+    return pv_info
+
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-usage-pv.py]: please remove the extra whitespace (3 empy lines) above here, leave one empty line between this and the next function

File: scripts/monitoring/cron-send-usage-pv.py [hunk 3]
```diff
@@ -102,6 +102,42 @@ def get_pv_usage():
...
+
+def convert_to_bytes(data):
+    """covert to ki"""
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-usage-pv.py]: """convert to bytes"""

File: scripts/monitoring/cron-send-usage-pv.py [hunk 4]
```diff
@@ -102,6 +102,42 @@ def get_pv_usage():
...
+        return int(number) * 1024
+    elif unit.lower() == 'mi':
+        return int(number) * 1024 * 1024
+    elif unit.lower() == 'gi':
+        return int(number) * 1024 * 1024 * 1024
+
+    raise Exception("invalid input data: " + data)
+
+def convert_to_kb(data):
+    """cover to kib"""
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-usage-pv.py]: convert

File: scripts/monitoring/cron-send-usage-pv.py [hunk 5]
```diff
@@ -102,6 +102,42 @@ def get_pv_usage():
...
+    return float(convert_to_bytes(data)) / 1024
+
+def convert_to_mb(data):
+    """cover to mb"""
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-usage-pv.py]: convert

File: scripts/monitoring/cron-send-usage-pv.py [hunk 6]
```diff
@@ -102,6 +102,42 @@ def get_pv_usage():
...
+    return float(convert_to_kb(data)) / 1024
+
+def convert_to_gb(data):
+    """covert to gb"""
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-usage-pv.py]: convert

**Review by drewandersonnz — CHANGES_REQUESTED** [review]
@BlueShells just a few "code flair" things to tidy, and can you please confirm the pv_used if it needs conversion or not?

File: scripts/monitoring/cron-send-usage-pv.py (hunk 1 above)
Comment (BlueShells) [reply scripts/monitoring/cron-send-usage-pv.py:167]: this don't needed , unless openshift changed in next version

**Review by drewandersonnz — APPROVED** [review]
+1

**Pull request merged**